main.ts

```diff
@@ -17,11 +17,13 @@ export default class NLSyntaxHighlightPlugin extends Plugin {
 
 		this.loadWordsToOverrideDict();
 
-		this.extensions = [NLSyntaxHighlightViewPlugin.extension];
+		this.extensions = this.settings.enabled ? [NLSyntaxHighlightViewPlugin.extension] : [];
 		this.registerEditorExtension(this.extensions);
-
+		
 		this.styleEl = document.head.createEl("style");
 		this.reloadStyle();
+
+		await this.addCommands();
```

Comment:
Since `addCommands()` is no longer async, we don't need an `await` here anymore :)

Reply:
Haha that's a bit embarrassing. Fixed